src/rustdoc-internals/search.md

```diff
@@ -354,3 +354,102 @@ The unification filter ensures that:
 The bloom filter checks none of these things,
 and, on top of that, can have false positives.
 But it's fast and uses very little memory, so the bloom filter helps.
+
+## Testing the search engine
+
+While the DOM UI code itself is tested using `rustdoc-gui` tests, the
+primary way the search engine is tested is the `rustdoc-js` and
+`rustdoc-js-std` tests. They run in NodeJS.
+
+A `rustdoc-js` test has a `.rs` and `.js` file, with the same name.
+The `.rs` file specifies the hypothetical library crate to run
+the searches on (make sure you mark anything you need to find as `pub`).
+The `.js` file specifies the actual searches.
+The `rustdoc-js-std` tests are the same, but don't require an `.rs`
+file, since they use the standard library.
+
+The `.js` file is like a module (except the loader takes care of
+`exports` for you). It expects you to set these variables in the
```

Comment:
the use of "expects" makes it sound like it is mandatory to set all of these (after all, `Option::expect` is used to throw an error if the receiver is `None`)

something like "It expects you to set *at least one of* these variables" or "the following variables have special meaning" would be more clear imo.